### PR TITLE
Fix : UTC based timezone

### LINKED ISSUE #2122: use UTC based timestamps
We use in mscolab `datetime.datetime.now() `
This can have a timezone issue, we should replace it by a timezone based on UTC
`datetime.datetime.now(tz=datetime.timezone.utc)`

### PR DESCRIPTION
**Pull request by Preetam-Das26**
**Purpose of PR?**:

Fixes #2122

**Does this PR introduce a breaking change?**
NO

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Does this PR results in some Documentation changes?**
_If yes, include the list of Documentation changes_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (Non-API breaking changes that adds functionality)
- [ ] PR Title follows the convention of  `<type>: <subject>`
- [ ] Commit has unit tests

### CODE REVIEW

**Comment by ReimarBauer** [comment]
please remove the pytest.log

File: .github/workflows/python-flake8.yml [hunk 1]
```diff
@@ -31,4 +31,5 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install flake8 flake8-builtins
+
```
Comment (ReimarBauer) [inline .github/workflows/python-flake8.yml]: ths change is not related to the issue

File: tests/_test_mscolab/test_file_manager.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def test_modify_user(self):
             user = User("user@example.com", "user", "password")
             assert user.id is None
             assert User.query.filter_by(emailid=user.emailid).first() is None
-            # creeat the user
+            # create the user
```
Comment (ReimarBauer) [inline tests/_test_mscolab/test_file_manager.py:68]: :) good catch

**Comment by ReimarBauer** [comment]
on "future" fixes you may be want to create a branch for an issue you work on.  git -b name

our stable branch means released on conda-forge.

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
remove accidantly committed files, line added to /python-flake8.yml

**Comment by Preetam-Das26** [comment]
Ok sure

**Comment by Preetam-Das26** [comment]
@ReimarBauer Kindly review the recent changes made .

**Review by ReimarBauer — APPROVED** [review]
Thx

**Comment by ReimarBauer** [comment]
not seen that I had to enable tests ;) let us see what they tell.

File: instance/mscolab.db
Comment (matrss) [inline instance/mscolab.db:1]: This does not belong in the repository.

File: .github/workflows/python-flake8.yml [hunk 3]
```diff
@@ -30,5 +30,5 @@ jobs:
     - name: Lint with flake8
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 flake8-builtins
+        pip install flake8 flake8-builtins
```
Comment (matrss) [inline .github/workflows/python-flake8.yml]: I think this happened by mistake.

**Review by matrss — CHANGES_REQUESTED** [review]

File: tests/_test_mscolab/test_file_manager.py [hunk 4]
```diff
@@ -74,12 +74,12 @@ def test_modify_user(self):
...
             assert self.fm.modify_user(user, action="create") is False
             # confirming the user
-            confirm_time = datetime.datetime.now() + datetime.timedelta(days=1)
+            confirm_time = datetime.datetime.now(tz=datetime.timezone.utc) + datetime.timedelta(days=1)
             self.fm.modify_user(user_query, attribute="confirmed_on", value=confirm_time)
             self.fm.modify_user(user_query, attribute="confirmed", value=True)
             user_query = User.query.filter_by(id=user.id).first()
             assert user_query.confirmed is True
-            assert user_query.confirmed_on == confirm_time
+            assert user_query.confirmed_on.replace(tzinfo=None) == confirm_time.replace(tzinfo=None)
```
Comment (matrss) [inline tests/_test_mscolab/test_file_manager.py:82]: I think this change hides a bug.

When I revert this I get the following error:
```
FAILED tests/_test_mscolab/test_file_manager.py::Test_FileManager::test_modify_user - assert datetime.datetime(2024, 2, 9, 8, 49, 39, 968400) == datetime.datetime(2024, 2, 9, 8, 49, 39, 968400, tzinfo=datetime.timezone.utc)
```

This means that the database query returns a naive datetime object, which is then not equal to the timezone aware object. I think the database query should return an aware datetime object instead, but I am not yet sure how to make it do that.

**Comment by matrss** [comment]
I also just noticed that we have a bunch of calls to `datetime.datetime.utcnow()`, which is even worse than `.now()` without a timezone. We should address those as well, but that could also be a follow-up PR.

File: tests/_test_mscolab/test_file_manager.py (hunk 4 above)
Comment (Preetam-Das26) [reply tests/_test_mscolab/test_file_manager.py:82]: Ok I didn't realise this situation. I will look into it.

Comment (matrss) [reply tests/_test_mscolab/test_file_manager.py:82]: This can also be handled in a follow-up PR because it is not exactly related to the initial issue that is addressed here. But I would prefer if this line stays as is (i.e. `assert user_query.confirmed_on == confirm_time`), and the test be marked as xfail instead to acknowledge that this is an issue.

**Review by matrss — APPROVED** [review]
